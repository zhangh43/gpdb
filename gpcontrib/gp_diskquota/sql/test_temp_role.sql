-- Test temp table restrained by role id
create schema strole;
create role u3temp nologin;
set search_path to strole;

select diskquota.set_role_quota('u3temp', '1MB');
create table a(i int);
alter table a owner to u3temp;
create temp table ta(i int);
alter table ta owner to u3temp;

-- expected failed: fill temp table
insert into ta select generate_series(1,100000000);
-- expected failed: 
insert into a select generate_series(1,100);
drop table ta;
select pg_sleep(5);
insert into a select generate_series(1,100);

drop table a;
reset search_path;
drop schema strole;
