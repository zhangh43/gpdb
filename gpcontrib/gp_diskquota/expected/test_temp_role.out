-- Test temp table restrained by role id
create schema strole;
create role u3temp nologin;
set search_path to strole;
select diskquota.set_role_quota('u3temp', '1MB');
 set_role_quota 
----------------
 
(1 row)

create table a(i int);
alter table a owner to u3temp;
create temp table ta(i int);
alter table ta owner to u3temp;
-- expected failed: fill temp table
insert into ta select generate_series(1,100000000);
ERROR:  role's disk space quota exceeded with name:u3temp
-- expected failed: 
insert into a select generate_series(1,100);
ERROR:  role's disk space quota exceeded with name:u3temp
drop table ta;
select pg_sleep(5);
 pg_sleep 
----------
 
(1 row)

insert into a select generate_series(1,100);
drop table a;
reset search_path;
drop schema strole;
